include/cuco/detail/bloom_filter/bloom_filter_impl.cuh

```diff
@@ -105,7 +105,7 @@ class bloom_filter_impl {
   __host__ constexpr void clear(cuda::stream_ref stream)
   {
     this->clear_async(stream);
-    stream.wait();
+    stream.sync();
```

Comment:
@miscco Thanks for working on this! This aligns with part of my plan in #695. It looks like the CI is failing due to the use of `stream.sync`. Has `wait` been deprecated in CCCL?

Reply:
I dropped the sync changes so we can get this merged